packages/create-cloudflare/src/__tests__/dialog.test.ts

```diff
@@ -0,0 +1,137 @@
+import { printSummary, printWelcomeMessage } from "../dialog";
+import type { C3Context } from "types";
+
+describe("dialog helpers", () => {
+	const std = collectCLIOutput();
+
+	test("printWelcomeMessage", () => {
+		printWelcomeMessage("0.0.0");
+
+		expect(normalizeOutput(std.out)).toMatchInlineSnapshot(`
+			" ╭──────────────────────────────────────────────────────────────╮
+			 │ 👋 Welcome to create-cloudflare v0.0.0!                      │
+			 │ 🧡 Let's get started.                                        │
+			 ╰──────────────────────────────────────────────────────────────╯
+			"
+		`);
+	});
+
+	describe("printSummary", () => {
+		const ctx: C3Context = {
+			project: { name: "test-project", path: "./workspace" },
+			args: {
+				projectName: "test-project",
+			},
+			template: {
+				configVersion: 1,
+				id: "test",
+				displayName: "display-name",
+				platform: "workers",
+			},
+			account: {
+				id: "account-id",
+				name: "account-name",
+			},
+			deployment: {
+				url: "https://example.test.workers.dev",
+			},
+			originalCWD: "./workspace",
+			gitRepoAlreadyExisted: false,
+		};
+
+		let originalStdoutColumns: number;
+
+		beforeAll(() => {
+			originalStdoutColumns = process.stdout.columns;
+		});
+
+		afterAll(() => {
+			process.stdout.columns = originalStdoutColumns;
+		});
+
+		test("with deploy", async () => {
+			await printSummary(ctx);
+
+			expect(normalizeOutput(std.out)).toMatchInlineSnapshot(`
+				" ╭───────────────────────────────────────────────────────────────────────────────────────╮
+				 │ 🎉  SUCCESS  Application deployed successfully!                                       │
+				 │                                                                                       │
+				 │ 🔍 View Project                                                                       │
+				 │    Visit: https://example.test.workers.dev                                            │
+				 │    Dash: https://dash.cloudflare.com/?to=/:account/workers/services/view/test-project │
+				 │                                                                                       │
+				 │ 💻 Continue Developing                                                                │
+				 │    Start dev server: npm run start                                                   │
+				 │    Deploy again: npm run deploy                                                      │
+				 │                                                                                       │
+				 │ 📖 Explore Documentation                                                              │
+				 │    https://developers.cloudflare.com/workers                                          │
+				 │                                                                                       │
+				 │ 💬 Join our Community                                                                 │
+				 │    https://discord.cloudflare.com                                                     │
```

Comment:
Is the alignment of the right hand side border correct here?

I understand emoji rendering on github may affect certain line widths but it looks like some other lines are off by a bit as well

Should we add an additional assertion that all lines must be the same number of chars? Might not be worth if if we have to use the Segmenter and special-case emoji width replacements

Reply:
It indeed has a different number of chars on lines without emoji even though it looks totally aligned on the terminal 😅 Looking into it why...

Reply:
It was because we normalized the package manager (from `pnpm` to `npm` ). So those 2 lines with the script will always be one characters less 😅 

I am gonna stop normalizing the package manager as we always run the unit tests in pnpm.